consensus/polybft/extra_test.go

```diff
@@ -0,0 +1,484 @@
+package polybft
+
+import (
+	"crypto/rand"
+	mrand "math/rand"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umbracle/fastrlp"
+)
+
+func TestExtra_Encoding(t *testing.T) {
+	parentStr := []byte("Here is the parent signature")
+	committedStr := []byte("Here is the committed signature")
+	bitmapStr := []byte("Here are the bitmap bytes")
+
+	addedValidators := newTestValidatorsWithAliases([]string{"A", "B", "C"}).getPublicIdentities()
+
+	removedValidators := bitmap.Bitmap{}
+	removedValidators.Set(2)
+
+	// different extra data for marshall/unmarshall
+	var cases = []struct {
+		extra *Extra
+	}{
+		{
+			&Extra{},
+		},
+		{
+			&Extra{
+				Validators: &ValidatorSetDelta{},
+				Parent:     &Signature{},
+				Committed:  &Signature{},
+			},
+		},
+		{
+			&Extra{
+				Validators: &ValidatorSetDelta{},
+				Seal:       []byte{3, 4},
+			},
+		},
+		{
+			&Extra{
+				Validators: &ValidatorSetDelta{
+					Added: addedValidators,
+				},
+				Parent:    &Signature{},
+				Committed: &Signature{},
+			},
+		},
+		{
+			&Extra{
+				Validators: &ValidatorSetDelta{
+					Removed: removedValidators,
+				},
+				Parent:    &Signature{AggregatedSignature: parentStr, Bitmap: bitmapStr},
+				Committed: &Signature{},
+			},
+		},
+		{
+			&Extra{
+				Validators: &ValidatorSetDelta{
+					Added:   addedValidators,
+					Removed: removedValidators,
+				},
+				Parent:    &Signature{},
+				Committed: &Signature{AggregatedSignature: committedStr, Bitmap: bitmapStr},
+			},
+		},
+		{
+			&Extra{
+				Parent:    &Signature{AggregatedSignature: parentStr, Bitmap: bitmapStr},
+				Committed: &Signature{AggregatedSignature: committedStr, Bitmap: bitmapStr},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		data := c.extra.MarshalRLPTo(nil)
+
+		extra := &Extra{}
+		assert.NoError(t, extra.UnmarshalRLP(data))
+		assert.Equal(t, c.extra, extra)
+	}
+}
+
+func TestExtra_UnmarshalRLPWith_NegativeCases(t *testing.T) {
```

Comment:
Test can be parallelized with `t.Parallel()`

Reply:
Fixed through #825